### PR TITLE
Config : support disabling failure persistence via env var

### LINKED ISSUE #276: Add an environment variable to disable failure persistence
As the title says, there should be an easy way to globally disable file failure persistence.

### PR DESCRIPTION
**Pull request by matthew-russo**
closes #276

### CODE REVIEW

File: proptest/src/test_runner/config.rs [hunk 1]
```diff
@@ -43,7 +43,10 @@ const FORK: &str = "PROPTEST_FORK";
 const TIMEOUT: &str = "PROPTEST_TIMEOUT";
 #[cfg(feature = "std")]
 const VERBOSE: &str = "PROPTEST_VERBOSE";
+#[cfg(feature = "std")]
```
Comment (matthew-russo) [inline proptest/src/test_runner/config.rs:46]: not strictly needed but noticed it was inconsistent

File: proptest/src/test_runner/config.rs [hunk 2]
```diff
@@ -217,7 +223,9 @@ pub struct Config {
     /// See the docs of [`FileFailurePersistence`](enum.FileFailurePersistence.html)
     /// and [`MapFailurePersistence`](struct.MapFailurePersistence.html) for more information.
     ///
-    /// The default cannot currently be overridden by an environment variable.
+    /// You can disable failure persistence with the `PROPTEST_DISABLE_FAILURE_PERSISTENCE`
+    /// environment variable but its not currently possible to set the persistence file
+    /// with an environment variable.
```
Comment (cameron1024) [inline proptest/src/test_runner/config.rs:228]: There was recently a PR adding a comment to these saying "the env var only has an effect when the `std` feature is enabled". I'm assuming that applies here? If so, should we add that warning?

**Review by cameron1024 — APPROVED** [review]
One minor thing, otherwise LGTM

File: proptest/src/test_runner/config.rs (hunk 2 above)
Comment (matthew-russo) [reply proptest/src/test_runner/config.rs]: Updated

**Review by rex-remind101 — APPROVED** [review]
looks good 👍

**Pull request merged**